modules/programs/msmtp.nix

```diff
@@ -0,0 +1,96 @@
+{ config, lib, pkgs, ... } @ top:
+
+with lib;
+with builtins;
+
+let
+  cfg = config.programs.msmtp;
+
+  sendMsmtpCommand = account:
+    if cfg.offlineSendMethod == "native" then
+      "${pkgs.msmtp}/bin/msmtpq -C $XDG_CONFIG_HOME/msmtp/config --account=${account.name} -t"
+    else
+      "${pkgs.msmtp}/bin/msmtp $XDG_CONFIG_HOME/msmtp/config --account=${account.name} -t";
+
+  # msmtp requires the password to finish with a newline
+  passwordCommandStr = account:
+    if account.passwordCommand != null then
+      ''passwordeval bash -c "echo $(${toString account.passwordCommand})"''
```

Comment:
actually msmtp can use libsecret without any directive which should be the preferred option and the one I use but for the sake of consistencym I tried to make passwordeval work. I might ask msmtp folks if we can deal without the newline.

Reply:
If it can use libsecret directly then that might go into a option (although I don't think it is necessary for now). In any case, I believe the `"${toString account.passwordCommand}; echo"` setup should work and be safe.